### PR TITLE
Use seg duration to time out remote transcoders.

### PR DESCRIPTION
**Pull request by j0sh**
**What does this pull request do? Explain your changes. (required)**

As described in https://github.com/livepeer/go-livepeer/issues/1478 , orchestrator timeouts for a remote T need to be adjusted to handle longer segment durations.

In addition to transmitting the duration from the B to the O in [`net.SegData`](https://github.com/livepeer/go-livepeer/commit/3358454a3099bdfbf747ca8e0071ed0f3f74713f#diff-afa2cdc3d0285595d3ca53db8b21226fR126) , the duration also needs to be made available to the orchestrator's [remote transcoder manager](https://github.com/livepeer/go-livepeer/commit/21568b10c3761fe80ed47bd03c4d576306feb725#diff-b76fbd82bbf5b7a4d2fdf59ecd8d4ca6R689) to make use of it. The remote transcoder manager follows the [Transcoder interface](https://github.com/livepeer/go-livepeer/commit/b1189c879a9962caac5df9e82b071c269a4d9265#diff-418dc657e12875665c93dffe96a0e464L23), which offers nowhere to put the duration:

```golang
Transcode(job string, fname string, profiles []ffmpeg.VideoProfile) (*TranscodeData, error)
```

Since both the orchestrator and broadcaster make use of `core.SegTranscodingMetadata` (CSTMD), which has similar fields as `net.SegData`, we formalize this relationship [1] and extract mappings ( [1](https://github.com/livepeer/go-livepeer/commit/445797907c0a692cccc1938b351190a31d65df26#diff-d8e92e957c82b6b03540a4be02e23720R255), [2](https://github.com/livepeer/go-livepeer/commit/445797907c0a692cccc1938b351190a31d65df26#diff-d8e92e957c82b6b03540a4be02e23720R494) ) to / from each: the latter for propagation over the wire, and the former for propagation throughout the codebase.

Having the `Transcoder` interface take a `CSTMD` effectively subsumes the `job` and `profiles` parameters, with `CSTMD.ManifestID` and `CSTMD.Profiles` respectively. From there, there is little reason not to also [incorporate the `fname`](https://github.com/livepeer/go-livepeer/commit/b9d39074ea3dd321d792725bc8470b26d755d670#diff-bc0ce8ba055f1f8945691e8ff8a6974dR26) parameter, resulting in the this  interface:

```
Transcode(md *SegTranscodingMetadata) (*TranscodeData, error)
```

The O-T protocol is also [adjusted to take `net.SegData`](https://github.com/livepeer/go-livepeer/commit/445797907c0a692cccc1938b351190a31d65df26#diff-afa2cdc3d0285595d3ca53db8b21226fR214). Although not strictly necessary since the standalone transcoder does not currently require the duration [2], this has the side effect of a small reduction in technical debt for standalone T, due to reuse of the mappings between `net.SegData` and `CSTMD` . Any fields that are added to these structs, such as a denominator for the framerate (https://github.com/livepeer/go-livepeer/issues/1474#issuecomment-630952970) will be available by default to standalone T. Back-compat logic that is developed within the mappings will also carry over.

[1] We could have opted to carry `net.SegData` throughout the codebase rather than making use of `CSTMD`, but the back-compat in `net.SegData` requires some translation before its fields can be used within the codebase, eg around profiles. This gives us the ability to evolve each structure as needed. We've already done so with the incorporation of the `Fname` field to CSTMD which isn't necessary over the wire.

[2] The transcoder will need the duration as soon as it's ready to be incorporated into the load balancer.

**Specific updates (required)**

- cmd, core: Typo fixes. 286a520
- net: Regenerate proto with segment duration. 5f03718
- core, server: Use net.SegData in the O-T protocol. … 4457979
- server, net: Send segment duration over the wire. 3358454
- server: Refactor test cases to use coreSegMetadata. … fe29742
- core: Add input filename to SegTranscodingMetadata. … b9d3907
- core, server: Pass SegTranscodingMetadata to transcoder. b1189c8
- core: Set remote transcoder timeout based on segment duration. 21568b1

**How did you test each of these updates (required)**

- Unit testing
- Manual testing

**Does this pull request close any open issues?**

**Checklist:**

- [ ] README and other documentation updated
- [ ] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

### CODE REVIEW

File: server/rpc.go
Comment (darkdarkdragon) [inline server/rpc.go]:
```suggestion
	dur := time.Duration(segData.Duration) * time.Millisecond
```

File: net/lp_rpc.proto
Comment (darkdarkdragon) [inline net/lp_rpc.proto:230]: recommended way of deprecating fields
```suggestion
    bytes profiles = 17 [deprecated=true];
```

File: net/lp_rpc.proto
Comment (darkdarkdragon) [inline net/lp_rpc.proto]:
```suggestion
  repeated VideoProfile fullProfiles = 33 [deprecated=true];
```

File: net/lp_rpc.proto
Comment (darkdarkdragon) [inline net/lp_rpc.proto]: recommended way of deprecating field that's not used anymore
```suggestion
    // string job      = 2;
    reserved 2;
    reserved "job";
```

**Review by darkdarkdragon — CHANGES_REQUESTED** [review]
Looks good, just couple of small suggestions

File: server/rpc.go
Comment (j0sh) [reply server/rpc.go]: Fixed in b7a6479c1ec41cf2a373d629eff8d1a687d40ec7

File: net/lp_rpc.proto
Comment (j0sh) [reply net/lp_rpc.proto:230]: Very nice - didn't know about this annotation! Actually, kept this line in place since we still "use" it via setting an invalid value to induce failures. Added a comment to the protobuf indicating this in 4d957856cdf5136a1e21972169e9f50d64e244c4

According to the protobuf docs, this is a no-op in all languages except Java, where it emits a warning. My concern is that if golang ever picks this up - golang only has errors, no warnings -  this could cause code to stop compiling and the annotation would have to be removed. I'm not entirely sure about the implications of this, so it seems safer to just keep things as-is.

File: net/lp_rpc.proto
Comment (j0sh) [reply net/lp_rpc.proto]: Made this `reserved` in 4d957856cdf5136a1e21972169e9f50d64e244c4

File: net/lp_rpc.proto
Comment (j0sh) [reply net/lp_rpc.proto]: Good idea, reserved the number in 4d957856cdf5136a1e21972169e9f50d64e244c4 . Didn't reserve the field name since we don't do JSON, so there should not be a conflict. (Can still add that in if you feel it should be there though.)

**Comment by j0sh** [comment]
Addressed comments via fixup, and also made a clean-up pass through the protobuf file to remove trailing whitespace in 98ab2120512f1dbad66da60fa0a784c93524db17. Cosmetic changes; nothing functional.

**Review by darkdarkdragon — APPROVED** [review]
LGTM

File: net/lp_rpc.proto
Comment (yondonfu) [inline net/lp_rpc.proto]: Needs an extra indent?

File: server/ot_rpc.go
Comment (yondonfu) [inline server/ot_rpc.go:141]: Perhaps we can short-circuit error handling by doing:

```golang
if err != nil {
    glog.Error("Unable to parse segData err=", err)
}
var tData *core.TranscodeData
if err == nil {
    tData, err = n.Transcoder.Transcode(md)
    ...
    if err == nil && len(tData.Segments) != len(profiles) {
        ...
    }
}
if err != nil {
    // Write error to body
}
```

Comment (j0sh) [reply server/ot_rpc.go:141]: Yeah it was the large reflow of code I was hoping to avoid, plus additional work to test the error code etc. Error handling is generally broken at this point, though; this isn't an isolated case. One of the larger tech-debty things in the O/T protocol.

File: net/lp_rpc.proto
Comment (j0sh) [reply net/lp_rpc.proto]: Was kind of intentional to indicate scope, "everything past this point is deprecated" but can re-indent

Comment (yondonfu) [reply net/lp_rpc.proto]: Got it. I don't feel strongly about the indentation so I'll leave that up to you.

File: server/ot_rpc.go
Comment (yondonfu) [reply server/ot_rpc.go:141]: Yeah the suggestion above was the least non-invasive change that came to mind, but definitely hear you that the structure could be cleaner/easier to test.

I'm fine with not going into a rabbit hole for this PR to fix all error handling in the O/T protocol. Created #1518 to track it - the description is pretty brief so feel free to add any relevant details there.

File: net/lp_rpc.proto
Comment (j0sh) [reply net/lp_rpc.proto]: Fixed indentation in 89a3acfe2b7155b203e0945682bb2b98f9e3192c and also improved the comment

File: server/ot_rpc.go
Comment (j0sh) [reply server/ot_rpc.go:141]: Thanks for making the ticket. Added a reference to the ticket in the code - 89a3acfe2b7155b203e0945682bb2b98f9e3192c

**Review by yondonfu — APPROVED** [review]
LGTM pending green CI

**Comment by j0sh** [comment]
Rebased, merged

**Pull request merged**
